# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build ffmpeg-build gst-plugins-base-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.15; lt-2.4.6)
libtoolize -c -f; aclocal -I m4 -I common/m4; autoconf; autoheader -f; automake -a -c
rm -rf gst-libs/ext
PKG_CONFIG_PATH=/x11/lib/pkgconfig LDFLAGS=-Wl,-rpath,/x11/lib ./configure --with-system-libav --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/lib/gstreamer-1.0
install ext/libav/.libs/libgstlibav.so $DESTDIR/x11/lib/gstreamer-1.0

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/gst-libav-build.pc << EOF
Name: gst-libav
Version: 1.10.5
Description: FFmpeg GStreamer plugin
Requires: $REQUIRES

exec=\\
/x11/lib/gstreamer-1.0/libgstlibav.so
EOF

exit
--- gst-libav-1.10.5.orig/configure.ac
+++ gst-libav-1.10.5/configure.ac
@@ -492,7 +492,6 @@
 ext/libav/Makefile
 ext/libswscale/Makefile
 gst-libs/Makefile
-gst-libs/ext/Makefile
 docs/Makefile
 docs/plugins/Makefile
 docs/version.entities
@@ -502,6 +501,9 @@
 pkgconfig/Makefile
 pkgconfig/gstreamer-plugins-libav-uninstalled.pc
 )
+if test "$with_system_libav" != "yes"; then
+  AC_CONFIG_FILES(gst-libs/ext/Makefile)
+fi
 AC_OUTPUT
 
 ORC_OUTPUT
